gradio/examples.py

```diff
@@ -198,7 +228,12 @@ def process_example(self, example_id: int) -> Tuple[List[Any], List[float]]:
             input_component.preprocess(raw_input[i])
             for i, input_component in enumerate(self.inputs)
         ]
-        predictions = self.fn(*processed_input)
+        if inspect.iscoroutinefunction(self.fn):
+            predictions = await self.fn(*processed_input)
+        else:
+            predictions = await anyio.to_thread.run_sync(
+                self.fn, *processed_input, limiter=self.limiter
+            )
```

Comment:
It looks like Example class is not linked to Blocks class, but could we make use of the function in Blocks?

`async def call_function(self, fn_index, processed_input):
`

Reply:
I agree it's similar code, but it's not possible to use `call_function()` as is. We would have to rewrite it to accept a function as an argument, which I think is overkill for this situation